README.md

```diff
@@ -29,12 +29,41 @@ A third party security audit was performed by Cure53, you can see the full repor
 
 `runc` only supports Linux. It must be built with Go version 1.21 or higher.
 
+### Pre-Requisites
+
+#### Go
+
 NOTE: if building with Go 1.22.x, make sure to use 1.22.4 or a later version
 (see [issue #4233](https://github.com/opencontainers/runc/issues/4233) for
 more details).
 
-In order to enable seccomp support you will need to install `libseccomp` on your platform.
-> e.g. `libseccomp-devel` for CentOS, or `libseccomp-dev` for Ubuntu
+#### Utilities and Libraries
+
+In addition to Go, building `runc` requires multiple utilities and libraries to be installed on your system.
+
+On Ubuntu/Debian, you can install the required dependencies with:
+
+```bash
+apt update && apt install -y make gcc libseccomp-dev pkg-config git
+```
+
+On CentOS/Fedora, you can install the required dependencies with:
+
+```bash
+yum install -y make gcc libseccomp-devel pkg-config git
```

Comment:
Can you verify building with only these on the respective container (like one for debian, another for ubuntu, fedora, etc.)?

It feels untested, given that alpine has linux-headers and here we don't. Am I missing something?

Reply:
Yup, as above, I will run a clean test.

Reply:
Yes, all works fine today in CI, locally, etc.

Reply:
So why isn't it working in a fresh ubuntu or fedora container? Is it a go configuration thing? A gcc configuration thing? Something else?

Can you try it?

```sh
docker run -it --rm -v $(pwd):/src -w /src ubuntu:22.04
```

Install go 1.22.5, install the various packages, in this case:

```sh
apt update && apt install -y make gcc libseccomp-dev pkg-config git
```

and then `make runc` or `make static`.

Reply:
Works fine here. I installed bash and golang from debian, and then installed go 1.22.5 from here (following this: https://go.dev/doc/install) because debian's version was too old.

Reply:
You did it in a debian container? Which tag? Do you mind pasting the output?

Reply:
I just recreated the same issue on `debian:sid-slim`. It is something common that you have set and I do not, or vice-versa.

Reply:
No, I did it in ubuntu, using the line you told me to use.

Here is the info about the ubuntu image:
```
$ docker inspect  08d22c0ceb15
[
    {
        "Id": "sha256:08d22c0ceb150ddeb2237c5fa3129c0183f3cc6f5eeb2e7aa4016da3ad02140a",
        "RepoTags": [
            "ubuntu:22.04"
        ],
        "RepoDigests": [
            "ubuntu@sha256:67211c14fa74f070d27cc59d69a7fa9aeff8e28ea118ef3babc295a0428a6d21"
        ],
```

Let me know if you need anything else to debug.

Reply:
This is really strange. I even went so far as to replace `gcc` with a script that dumps the call to `/tmp/gcc.out` just so I can see what is called differently.

Reply:
OK, looks like you were onto something, but not sure. I ran the command with `go build -x`. What surprised me was this output:

```sh
go build -x -trimpath "-buildmode=pie"  -tags "seccomp urfave_cli_no_docs " -ldflags "-X main.gitCommit=v1.2.0-rc.2-22-g41
9ef873 -X main.version=1.2.0-rc.2+dev " -o runc .
WORK=/tmp/go-build2257789363
cd /pwd
git status --porcelain
cd /pwd
git -c log.showsignature=false show -s --format=%H:%ct
mkdir -p $WORK/b141/
cd /usr/local/go/src/runtime/cgo
TERM='dumb' CGO_LDFLAGS='"-O2" "-g" "-lpthread"' /usr/local/go/pkg/tool/linux_amd64/cgo -objdir $WORK/b141/ -importpath runtime/cgo -import_runtime_cgo=false -import_syscall=false -- -I $WORK/b141/ -O2 -g -Wall -Werror -fno-stack-protector ./cgo.go
# runtime/cgo
gcc-13: error: unrecognized command-line option '-m64'
```

Why would it be running `/usr/local/go/pkg/tool/linux_amd64/cgo`? That is for building for amd64. Yet, I am running on aarch64. 

GOARCH is unset. If I execute it with `GOARCH=arm64`, then it works, e.g.:

```
make static GOARCH=arm64
```

Oh crap! It is my fault! I copied the wrong link for installing go, had the two of them - amd64 and arm64 - and copied the wrong one! 🤯

Reply:
I can confirm that the builds are not an issue now. Alpine needs `linux-headers` added explicitly, the others do not.

Reply:
Yeah, and what a silly mystery. Sigh. Oh well, may all our problems be that easy to solve.

> We are not targetting building inside a container (or not only), I think all the dependencies should be listed. IMHO we mention the respective package for each distro.

Agreed. It was just easier to test the various distros in the container.

No, the system doesn't come with it:

```sh
$ docker run -it --rm -v $(pwd):/pwd -w /pwd ubuntu:22.04
# ls -l /usr/include/linux/stat.h
ls: cannot access '/usr/include/linux/stat.h': No such file or directory
```

However, installing gcc causes it to exist:

```sh
# apt install -y gcc
...
# ls -l /usr/include/linux/stat.h 
-rw-r--r-- 1 root root 6726 Jul  1 09:13 /usr/include/linux/stat.h
# dpkg-query -S /usr/include/linux/stat.h
linux-libc-dev:arm64: /usr/include/linux/stat.h
```

That isn't going to be fun to track down, but I don't think we need to. As long as we explicitly say you need `linux-libc-dev` - which makes sense, and parallels `musl-dev` on alpine - we should be good.

For fedora, it also is `gcc`, which installs `kernel-headers`, so we can do that package there.

```sh
# rpm -qf /usr/include/linux/stat.h
kernel-headers-6.9.4-200.fc40.aarch64
```

I updated the doc to reflect it. Take a look.

Reply:
Thanks!